api/v1alpha1/etcd_types.go

```diff
@@ -42,19 +42,18 @@ type MetricsLevel string
 type GarbageCollectionPolicy string
 
 // StorageProvider defines the type of object store provider for storing backups.
-// +kubebuilder:validation:Enum=aws;gcp;os;azure;alicloud
 type StorageProvider string
 
 // StoreSpec defines parameters related to ObjectStore persisting backups
 type StoreSpec struct {
-	// +optional
-	Container string `json:"container,omitempty"`
+	// +required
+	Container string `json:"container"`
```

Comment:
Why are we making them required?  They were optional considering the case on no backup i.e. empty provider and 'Local' provider for development case.

Reply:
changed.